tests/e2e/main.go

```diff
@@ -49,29 +50,71 @@ var (
 )
 
 var (
-	testSelection TestSet
-	testMap       map[string]*testRunWithSteps = map[string]*testRunWithSteps{
-		"happy-path-short": {
-			testRun: DefaultTestRun(), steps: shortHappyPathSteps,
-			description: `This is like the happy path, but skips steps
-that involve starting or stopping nodes for the same chain outside of the chain setup or teardown.
-This is suited for CometMock+Gorelayer testing`,
-		},
-		"light-client-attack": {
-			testRun: DefaultTestRun(), steps: lightClientAttackSteps,
-			description: `This is like the short happy path, but will slash validators for LightClientAttackEvidence instead of DuplicateVoteEvidence.
-This is suited for CometMock+Gorelayer testing, but currently does not work with CometBFT,
-since causing light client attacks is not implemented.`,
-		},
-		"happy-path":       {testRun: DefaultTestRun(), steps: happyPathSteps, description: "happy path tests"},
-		"changeover":       {testRun: ChangeoverTestRun(), steps: changeoverSteps, description: "changeover tests"},
-		"democracy-reward": {testRun: DemocracyTestRun(true), steps: democracySteps, description: "democracy tests allowing rewards"},
-		"democracy":        {testRun: DemocracyTestRun(false), steps: rewardDenomConsumerSteps, description: "democracy tests"},
-		"slash-throttle":   {testRun: SlashThrottleTestRun(), steps: slashThrottleSteps, description: "slash throttle tests"},
-		"multiconsumer":    {testRun: MultiConsumerTestRun(), steps: multipleConsumers, description: "multi consumer tests"},
+	selectedTests TestSet
+	testRuns      = map[string]TestRunChoice{
+		"default":          {name: "default", testRun: DefaultTestRun(), description: "default test run"},
+		"changeover":       {name: "changeover", testRun: ChangeoverTestRun(), description: "changeover test run"},
+		"democracy":        {name: "democracy", testRun: DemocracyTestRun(false), description: "democracy test run"},
+		"democracy-reward": {name: "democracy-reward", testRun: DemocracyTestRun(true), description: "democracy test run with rewards"},
+		"slash-throttle":   {name: "slash-throttle", testRun: SlashThrottleTestRun(), description: "slash throttle test run"},
+		"multiconsumer":    {name: "multiconsumer", testRun: MultiConsumerTestRun(), description: "multi consumer test run"},
 	}
+	// helper function to get the test run choices by matching test runs
 )
 
+var selectedTestfiles TestSet
+
+var stepChoices = map[string]StepChoice{
```

Comment:
Separate step choices from test runs.

Why: When we input files, we need to give the test runs separately anyways. This makes that behaviour universal. Also seems more future proof, e.g. for running the same steps with different test runs/contexts

Reply:
To a semi outsider like myself, the abstractions here are not super intuitive. 

"steps" seems like a set of steps to make in running a test.

"test runs" seems synonymous to "steps"